python/kserve/kserve/model.py

```diff
@@ -174,47 +177,55 @@ def postprocess(self, response: Union[Dict, ModelInferResponse]) -> Dict:
             return response.get_response(as_json=True)
         return response
 
-    async def _http_predict(self, request: Dict) -> Dict:
+    async def _http_predict(self, payload: Dict, headers: Dict[str, str] = None) -> Dict:
         predict_url = PREDICTOR_URL_FORMAT.format(self.predictor_host, self.name)
         if self.protocol == PredictorProtocol.REST_V2.value:
             predict_url = PREDICTOR_V2_URL_FORMAT.format(self.predictor_host, self.name)
-        json_header = {'Content-Type': 'application/json'}
+
+        # Adjusting headers. Inject content type if not exist.
+        # Also, removing host, as the header is the one passed to transformer and contains transformer's host
+        if not ('Content-Type' in headers):
+            headers['Content-Type'] = 'application/json'
+        headers.pop("Host", None)
         response = await self._http_client.fetch(
             predict_url,
             method='POST',
             request_timeout=self.timeout,
-            headers=json_header,
-            body=json.dumps(request)
+            headers=headers,
```

Comment:
The headers are from the original request, I think we need to explicitly to copy over the selected headers if needed for the new request.

Reply:
But what headers should we copy over unless we configure allowed headers somewhere beforehand? Also, wouldn't someone need trace id to track any requests in their system for error logging and debugging until we have distributed tracing properly implemented?
For example any wrong inputs, etc can be traced easily if the trace id goes across transformer, predictor, explainer.

Reply:
Headers like `X-RequestId` and TraceId should be copied over as they are the same on the request chain, however request specific headers like `Content-Length`, `SpanId`, `Host` should not be propagated. I think explicitly adding the headers required for propagation is easier to reason about than copying by default which can be hard to debug if some header is passed along unexpectedly like the `Host` issue here.

Reply:
@yuzisun  @andyi2it : We should have some way of configuring the headers that we wish to be propagated. Because I believe most of the companies have their own trace-id in their ecosystem and they would want that to be present along the chain.
By just retaining over the 2 headers  ['X-Request-Id', 'X-B3-Traceid'], we are assuming that no additional headers can be there. 

Isn't it a blocker for adoption by the community? 

For a long term solution, I believe we can have a feature to support configuring headers that users want to propagate.
For short term solution , are we open to extend this [if section](https://github.com/kserve/kserve/blob/master/python/kserve/kserve/model.py#L187-L191) to at least include more generic headers like `Authorization` ?